xdsl/dialects/builtin.py

```diff
@@ -368,8 +372,76 @@ def size(self) -> int:
         return (self.bitwidth + 7) >> 3
 
 
+_PyT = TypeVar("_PyT")
+
+
+class PackableType(Generic[_PyT], FixedBitwidthType, ABC):
```

Comment:
Do we want to combine the `FixedBitwidthType` and `PackableType`? Maybe we are mixing host bitdwidth with target bitwidth? The target system could for example have support for `i4` types, but will be packed as `i8` at compile time.

Maybe this is too niche, I'm just not to sure about this one, the two concepts just seem to have little in common

Or `f80` or `f128` for example, they are not packable, but do have a fixed bitwidth

Reply:
Starting to change my mind about this, if something has a fixedbitwidthtype it should be packable, just not necesarrily packable with that specified fixed bitwidth. I guess it thus makes sense to keep them together